integrations/dd-sdk-android-okhttp/src/test/kotlin/com/datadog/android/okhttp/trace/TracingInterceptorTest.kt

```diff
@@ -1659,6 +1659,117 @@ internal open class TracingInterceptorTest {
         verify(mockFuture).get(1, TimeUnit.SECONDS)
     }
 
+    @Test
+    fun `M prevent infinite loop W listener retries same request multiple times`(
+        @IntForgery(min = 200, max = 300) statusCode: Int
+    ) {
+        // Given
+        var interceptCallCount = 0
+        val maxRetries = TracingInterceptor.MAX_REQUEST_INTERCEPTION_COUNT
+
+        whenever(mockResolver.isFirstPartyUrl(fakeUrl.toHttpUrl())).thenReturn(true)
+        stubChain(mockChain, statusCode)
+
+        whenever(
+            mockRequestListener.onRequestIntercepted(any(), any(), anyOrNull(), anyOrNull())
+        ).doAnswer {
+            interceptCallCount++
+            if (interceptCallCount <= maxRetries + 1) {
+                testedInterceptor.intercept(mockChain)
+            }
+            return@doAnswer Unit
+        }
+
+        // When
+        testedInterceptor.intercept(mockChain)
+
+        // Then
+        verify(mockRequestListener, times(maxRetries)).onRequestIntercepted(any(), any(), anyOrNull(), anyOrNull())
+
+        verify(mockInternalLogger).log(
+            eq(InternalLogger.Level.WARN),
+            eq(InternalLogger.Target.USER),
+            any(),
```

Comment:
we need to check that the exact message is being printed. Look for `verifyLog` method across other tests